api_app/core/views.py

```diff
@@ -54,9 +61,41 @@ def kill(self, request, job_id, name):
         # kill celery task
         celery_app.control.revoke(report.task_id, terminate=True)
         # update report
-        report.status = AbstractReport.Statuses.KILLED.name
-        report.save(update_fields=["status"])
+        report.update_status(AbstractReport.Statuses.KILLED.name)
         # execute callback post kill
         self._post_kill(report)
```

Comment:
i'd say that this part of the code should be a method named `perform_kill()`   (if you see the DRF code, that's the naming convention they follow). Then we do not require `_post_kill` since we can simply override the method in the subclasses and call `super()`.

Reply:
Similar change could be made for `perform_retry` where we can simply call `super()` and then subsequent individual logic.